posthog/batch_exports/http.py

```diff
@@ -167,51 +156,43 @@ class Meta:
             "last_paused_at",
             "start_at",
             "end_at",
-            "trigger_immediately",
-        ]
-        read_only_fields = [
-            "id",
-            "paused",
-            "created_at",
-            "last_updated_at",
+            "latest_runs",
         ]
+        read_only_fields = ["id", "created_at", "last_updated_at", "latest_runs"]
 
     def create(self, validated_data: dict) -> BatchExport:
         """Create a BatchExport."""
         destination_data = validated_data.pop("destination")
         team_id = self.context["team_id"]
-        interval = validated_data.pop("interval")
-        name = validated_data.pop("name")
-        start_at = validated_data.get("start_at", None)
-        end_at = validated_data.get("end_at", None)
-        trigger_immediately = validated_data.get("trigger_immediately", False)
-
-        return create_batch_export(
-            team_id=team_id,
-            interval=interval,
-            name=name,
-            destination_data=destination_data,
-            start_at=start_at,
-            end_at=end_at,
-            trigger_immediately=trigger_immediately,
-        )
 
-    def update(self, instance: BatchExport, validated_data: dict) -> BatchExport:
+        destination = BatchExportDestination(**destination_data)
+        batch_export = BatchExport(team_id=team_id, destination=destination, **validated_data)
+
+        sync_batch_export(batch_export, created=True)
+
+        with transaction.atomic():
+            destination.save()
+            batch_export.save()
+
+        return batch_export
+
+    def update(self, batch_export: BatchExport, validated_data: dict) -> BatchExport:
         """Update a BatchExport."""
         destination_data = validated_data.pop("destination", None)
-        interval = validated_data.get("interval", None)
-        name = validated_data.get("name", None)
-        start_at = validated_data.get("start_at", None)
-        end_at = validated_data.get("end_at", None)
-
-        return update_batch_export(
-            batch_export=instance,
-            interval=interval,
-            name=name,
-            destination_data=destination_data,
-            start_at=start_at,
-            end_at=end_at,
-        )
+
+        with transaction.atomic():
+            if destination_data:
+                batch_export.destination.type = destination_data.get("type", batch_export.destination.type)
+                batch_export.destination.config = {
+                    **batch_export.destination.config,
+                    **destination_data.get("config", {}),
+                }
```

Comment:
The update tests are failing. Is it possible this is missing a call to `save` the model?

Reply:
It could be yeah... I wasn't sure if Django took care of saving the relational data as well but it seems like it doesn't...